include/inexor/vulkan-renderer/wrapper/query_pool.hpp

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <vulkan/vulkan_core.h>
+
+#include <string>
+#include <vector>
+
+namespace inexor::vulkan_renderer::wrapper {
+
+// Forward declaration of device wrapper
+class Device;
+
+/// @brief A wrapper for Vulkan query pools
+class QueryPool {
+private:
+    const Device &m_device;
+    VkPhysicalDeviceFeatures m_device_features{};
+    VkQueryPool m_query_pool{};
+    std::vector<std::uint64_t> m_pipeline_stats;
+    std::vector<std::string> m_pipeline_stat_names;
+
+    /// @brief These pipeline statistics are enabled by default if the default constructor is used.
+    /// @note We are not storing these as VkQueryPipelineStatisticFlags, because we need to perform additional checks
+    /// for some of these flags in order to use them. For example we need to check if tessellation is enabled in order
+    /// to query its performance. Please note computer shaders do not require special checks.
+    /// <a href="https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#queries-pipestats">
```

Comment:
So yeah we can generate Doxygen which contains links to the official Vulkan spec. The line looks a little long but I think it's fine.

Reply:
Nope. It has problem with the hashtag in the link.

Reply:
Maybe there is a way of masking it though?

Reply:
I will keep this for now.